### PR TITLE
Ignore iptable delete failures for nonexistent rules

### PR DESCRIPTION
**Pull request by jaellio**
**Please provide a description of this PR:**

TODO: Add additional test in `iptables_e2e_test,go`

Please consider much simpler alternative: https://github.com/istio/istio/commit/1b03f56e1476cf850afe11f70abdb15ed810e686

### CODE REVIEW

File: cni/pkg/iptables/iptables.go [hunk 1]
```diff
@@ -175,7 +175,12 @@ func (cfg *IptablesConfigurator) executeDeleteCommands() error {
 
 	for _, iptVer := range iptablesVariant {
 		for _, cmd := range deleteCmds {
-			delErrs = append(delErrs, cfg.ext.Run(iptablesconstants.IPTables, &iptVer, nil, cmd...))
+			err := cfg.ext.Run(iptablesconstants.IPTables, &iptVer, nil, cmd...)
+			if err != nil && strings.Contains(err.Error(), "No chain/target/match by that name") {
```
[the remark this answers is not in the document]
Comment (jaellio) [reply cni/pkg/iptables/iptables.go:179]: I am fine with your simplified approach. I had already made these changes when I saw you comment so I thought I push go ahead and push them for discussion :)

File: cni/pkg/iptables/iptables.go [hunk 2]
```diff
@@ -423,10 +428,20 @@ func (cfg *IptablesConfigurator) executeCommands(log *istiolog.Scope, iptablesBu
 	var execErrs []error
 
 	// Execute iptables-restore
-	execErrs = append(execErrs, cfg.executeIptablesRestoreCommand(log, iptablesBuilder.BuildV4Restore(), &cfg.iptV))
+	err := cfg.executeIptablesRestoreCommand(log, iptablesBuilder.BuildV4Restore(), &cfg.iptV)
+	if err != nil && strings.Contains(err.Error(), "Chain already exists") {
```
[the remark this answers is not in the document]
Comment (jaellio) [reply cni/pkg/iptables/iptables.go:432]: Good callout about "chain already exists" not being safe to ignore. To clarify, without https://github.com/istio/istio/pull/53153 are reruns of the istio-iptables not supported?

**Comment by jaellio** [comment]
Closing in favor of https://github.com/istio/istio/pull/53951

File: cni/pkg/iptables/iptables.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (leosarra) [reply cni/pkg/iptables/iptables.go]: Correct, no kind of reruns is supported for CNI iptables at the moment.
Sidecar/VM deployments on the other end already supports idempotency (which means equivalent reruns can be performed) and reconciliation (which is disabled by default, unequal reruns are not supported out-of-the-box) as it was added by https://github.com/istio/istio/pull/50328/

**Pull request closed without merging**